src/AzureStorage/Queues/AzureStorageQueue.cs

```diff
@@ -153,17 +169,17 @@ public class AzureStorageQueue<T> : QueueBase<T> where T : class {
             _workerErrorCount = 0;
         }
 
-        public override void StartWorking(Func<IQueueEntry<T>, CancellationToken, Task> handler, bool autoComplete = false, CancellationToken cancellationToken = new CancellationToken()) {
+        protected override void StartWorkingImpl(Func<IQueueEntry<T>, CancellationToken, Task> handler, bool autoComplete = false, CancellationToken cancellationToken = new CancellationToken()) {
             if (handler == null)
                 throw new ArgumentNullException(nameof(handler));
-
+            
             var linkedCancellationToken = CancellationTokenSource.CreateLinkedTokenSource(_queueDisposedCancellationTokenSource.Token, cancellationToken).Token;
 
             Task.Run(async () => {
```

Comment:
@ejsmith The benefit is we can call `EnsureQueueCreatedAsync` without `.WaitAndUnwrapException()` - `Task.Run` would still be required as a fire and forget (except service bus which has an `OnMessageAsync` callback).